packages/core/src/StateMachine.ts

```diff
@@ -446,58 +446,64 @@ export class StateMachine<
   }
 
   public restoreState(
-    state: State<TContext, TEvent, TResolvedTypesMeta>,
+    state: PersistedMachineState<any>,
     _actorCtx?: ActorContext<
       TEvent,
       State<TContext, TEvent, TResolvedTypesMeta>
     >
   ): State<TContext, TEvent, TResolvedTypesMeta> {
-    let restoredState: State<TContext, TEvent, TResolvedTypesMeta>;
+    const self = this;
```

Comment:
oh wow, this is a real blast from the past 😅 you are already using an arrow function below - you don't need this old school trick, just use `this`

Reply:
Yep, there was a point during development where I need to do this, accidentally left it in hah